app/models/hood.rb

```diff
@@ -2,11 +2,41 @@ class Hood < ApplicationRecord
   has_many :houses
   has_many :home_owners_associations
 
+  def valuation_before(date)
+    house_count = houses.count
+
+    # keep track of dates, and how many houses had a price at that point in time
+    # default number of houses to 0
```

Comment:
This algorithm finds the valuation of a neighborhood at a date when ALL the houses were valuation. This ensures that the valuation is an exact number at that date as opposed to an estimate.

Another approach we can do is add up the latest valuation of each house prior to the specified date, though this might give us an estimate/a less accurate valuation of the neighborhood. This would simplify the code quite a bit, but it loses accuracy.